plugins/application/elasticsearch/main.go

```diff
@@ -126,16 +126,20 @@ func (es *Elasticsearch) Run(ctx context.Context, done chan bool) {
 		es.logger.Info("removed indices")
 	}
 
-	for {
-		select {
-		case <-ctx.Done():
-			goto done
-		case dumped := <-es.dump:
-			if err := es.client.Index(dumped.index, dumped.record, es.configuration.BulkIndex); err != nil {
-				es.logger.Metadata(logging.Metadata{"plugin": appname, "event": dumped.record, "error": err})
-				es.logger.Error("failed to index event - disregarding")
-			} else {
-				es.logger.Debug("successfully indexed document(s)")
+	for i := 0; i < es.configuration.IndexWorkers; i++ {
+		es.logger.Metadata(logging.Metadata{"plugin": appname, "worker-id": i})
+		es.logger.Debug("spawning ES API worker")
+		for {
```

Comment:
As discussed, probably missing a `go func{ }` around this for loop

Reply:
Fixed.